### PR TITLE
feat(migrate): Add path conventions for package configuration files

### PR DESCRIPTION
**Pull request by mnonnenmacher**
Apply the path conventions from the ort-config repository also to package configuration files.

### CODE REVIEW

File: plugins/commands/migrate/src/main/kotlin/MigrateCommand.kt [hunk 1]
```diff
@@ -169,3 +182,12 @@ private fun convertHoconToYaml(hocon: String): String {
...
 }
+
+/**
+ * Return the path for the directory to store [PackageConfiguration]s for this [Identifier].
+ *
+ * TODO: Replace with equivalent function from the package configuration provider for the ort-config repository, once
+ *       implemented.
+ */
+fun Identifier.toPackageConfigurationDir() =
+    "${type.encodeOr("_")}/${namespace.encodeOr("_")}/${name.encodeOr("_")}/${version.encodeOr("_")}"
```
Comment (sschuberth) [inline plugins/commands/migrate/src/main/kotlin/MigrateCommand.kt:193]: Can be shortened to `toPath(emptyValue = "_")` like done in the [*helper-cli*'s `CreateCommand` for package configurations](https://github.com/oss-review-toolkit/ort/blob/5a92da88ce2176de3bb9e979a39a3e54cecb2282/helper-cli/src/main/kotlin/commands/packageconfig/CreateCommand.kt#L152C25-L152C50).

**Review by sschuberth — CHANGES_REQUESTED** [review]

File: plugins/commands/migrate/src/main/kotlin/MigrateCommand.kt [hunk 2]
```diff
@@ -169,3 +182,12 @@ private fun convertHoconToYaml(hocon: String): String {
...
     return yamlLines.joinToString("\n")
 }
+
+/**
+ * Return the path for the directory to store [PackageConfiguration]s for this [Identifier].
+ *
+ * TODO: Replace with equivalent function from the package configuration provider for the ort-config repository, once
+ *       implemented.
+ */
+fun Identifier.toPackageConfigurationDir() =
```
Comment (fviernau) [inline plugins/commands/migrate/src/main/kotlin/MigrateCommand.kt:192]: Instead of this function, could you use `Identifier.toPath(emptyValue = "_")`?

**Review by fviernau — CHANGES_REQUESTED** [review]

File: plugins/commands/migrate/src/main/kotlin/MigrateCommand.kt (hunk 2 above)
Comment (sschuberth) [reply plugins/commands/migrate/src/main/kotlin/MigrateCommand.kt:192]: Isn't this the same as my comment above?

Comment (fviernau) [reply plugins/commands/migrate/src/main/kotlin/MigrateCommand.kt:192]: It is yes. I haven't seen it. Maybe I didn't hit F5 right before the review.

**Review by sschuberth — APPROVED** [review]

**Review by fviernau — APPROVED** [review]
LGTM, but unused imports are left....

**Review by sschuberth — APPROVED** [review]

**Review by fviernau — APPROVED** [review]

**Pull request merged**